File: AndroidLearnProject/app/src/main/code_backup/mpresenter/WechatPresenter.java
package com.android.learn.mpresenter;


import com.android.learn.base.mmodel.BannerListData;
import com.android.learn.base.mmodel.BaseData;
import com.android.learn.base.mmodel.BaseResponse;
import com.android.learn.base.mmodel.FeedArticleListData;
import com.android.learn.base.mmodel.FeedArticleListData.FeedArticleData;
import com.android.learn.base.mmodel.WxArticle;
import com.android.learn.base.mpresenter.BasePresenter;
import com.android.learn.base.thirdframe.rxjava.BaseObserver;
import com.android.learn.base.utils.ResponseStatusUtil;
import com.android.learn.mcontract.HomeContract;
import com.android.learn.mcontract.WechatContract;

import java.util.List;

import io.reactivex.Observable;


public class WechatPresenter extends BasePresenter<WechatContract.View> implements WechatContract.Presenter {

    @Override
    public void getWxArticle() {
        Observable observable = mRestService.getWxArticle();
        addSubscribe(observable, new BaseObserver<BaseResponse<List<WxArticle>>>(false) {
            @Override
            public void onNext(BaseResponse<List<WxArticle>> datas) {
                mView.showWxArticle(datas.data);
            }
        });
    }

}

File: AndroidLearnProject/app/src/main/java/com/android/learn/mpresenter/WechatPresenter.kt
package com.android.learn.mpresenter


import com.android.learn.base.mmodel.BannerListData
import com.android.learn.base.mmodel.BaseData
import com.android.learn.base.mmodel.BaseResponse
import com.android.learn.base.mmodel.FeedArticleListData
import com.android.learn.base.mmodel.FeedArticleListData.FeedArticleData
import com.android.learn.base.mmodel.WxArticle
import com.android.learn.base.mpresenter.BasePresenter
import com.android.learn.base.thirdframe.rxjava.BaseObserver
import com.android.learn.base.utils.ResponseStatusUtil
import com.android.learn.mcontract.HomeContract
import com.android.learn.mcontract.WechatContract

import io.reactivex.Observable


class WechatPresenter : BasePresenter<WechatContract.View>(), WechatContract.Presenter {

    override fun getWxArticle() {
        val observable = mRestService.wxArticle
        addSubscribe(observable, object : BaseObserver<BaseResponse<List<WxArticle>>>(false) {
            override fun onNext(datas: BaseResponse<List<WxArticle>>) {
                mView!!.showWxArticle(datas.data)
            }
        })
    }

}
